runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -12,9 +12,21 @@ TSHighlighterQuery.__index = TSHighlighterQuery
 
 local ns = a.nvim_create_namespace("treesitter/highlighter")
 
+local AtomStyleCaptureFallback = {}
+
+-- If @definition.special does not exist use @definition instead
+function AtomStyleCaptureFallback:__index(capture)
+  local rtn
+  while capture:find('.', 1, true) and not rtn do
+    capture = capture:gsub('%..*$', '')
+    rtn = rawget(self, capture)
```

Comment:
I did not add the new capture to the highlight map for the case that the content of this dictionary changes.

Reply:
Probably I should use `string:split`

Reply:
I think we should cache it somehow (by either adding back to the map or another map). This lookup happens A LOT.

Reply:
It is now cached